src/librustc_resolve/lib.rs

```diff
@@ -1855,48 +1858,46 @@ impl<'a> Resolver<'a> {
                     .collect(),
             }
         };
-        let res = self.resolve_ast_path_cb(&path, is_value, |_, _, _| errored = true);
-        if errored || res == def::Res::Err {
-            Err(())
-        } else {
-            Ok((path, res))
+        match self.resolve_ast_path_inner(&path, is_value) {
+            Ok(res) => {
```

Comment:
```suggestion
            Ok(res) if res != Res::Err => {
```
Although, I'd rather remove this condition (and then use `?`).
If the function returns `Res`, then rustdoc should be ready to deal with `Res::Err` as well.